### PR TITLE
Fixed Deferred Fog does not work in XR single pass

### PR DESCRIPTION
**Pull request by thomas-zeng**
---
### Purpose of this PR
- Fixed fogbugz case https://fogbugz.unity3d.com/f/cases/1390236/

---
### Testing status
- QA
- ABV

---
### Comments to reviewers
- Added missing macros.

### CODE REVIEW

**Review by phi-lira — APPROVED** [review]

**Comment by thomas-zeng** [comment]
I believe the failed tests are not related to the code changes(shader only changes).

**Comment by thomas-zeng** [comment]
Hello @ryanhy-unity , could you help testing this PR real quick? Should be a low risk PR as it only touches shaders. I have tested the repro project on PC DX11 and verified the fix works. Would be nice to get some on device coverage.

File: com.unity.render-pipelines.universal/CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed material converter not being able to be called in batch mode. [case 1375962]
 - Fixed an issue where specular color was not matching behaviour in Legacy and HDRP. [case 1326941](https://issuetracker.unity3d.com/issues/urp-specular-color-behavior-does-not-match-legacy-or-hdrp)
 - Fixed a shader compiler issue with mismatching variable types when calling lerp.
+- Fixed URP Deferred Fog pass does not work in XR singlepass. [case 1390236]
```
Comment (sandy-carter-unity) [inline com.unity.render-pipelines.universal/CHANGELOG.md:26]: Link is incomplete, should be 
```md
[case 1390236](https://issuetracker.unity3d.com/product/unity/issues/guid/1390236/)
```

**Review by sandy-carter-unity — APPROVED** [review]
LGTM

File: com.unity.render-pipelines.universal/CHANGELOG.md (hunk 1 above)
Comment (thomas-zeng) [reply com.unity.render-pipelines.universal/CHANGELOG.md]: Addressed in [06d211e]

**Review by phi-lira — APPROVED** [review]

**Pull request merged**